app/models/weather.rb

```diff
@@ -22,6 +22,25 @@ def chance_of_rain(day_index)
     raw_day["precipProbability"] * 100
   end
 
+  def precip_type(day_index)
+    type = weather_info["daily"]["data"][day_index]["precipType"]
+    if type == nil
+      type = "precipitation"
+    end
+    type
+  end
+
+  def precip_icon(day_index)
+    type = weather_info["daily"]["data"][day_index]["precipType"]
```

Comment:
This is an opportunity for refactor if you wanted since you're using the same line here and line 26. Could have a method called "weather_type" or similar that just returns this line. 
Then you wouldn't have to save it as a variable in your methods and could just use it directly in your conditionals. 
Just an option, wouldn't say it's necessary.

Reply:
I definitely agree with going down this route. For now, I'm going to make the other suggested changes and push it up. (As time is getting short and this branch has been tricky, and I'd like it merged into master and tested on Heroku asap).